autopush/websocket.py

```diff
@@ -45,6 +45,33 @@ class SimplePushServerProtocol(WebSocketServerProtocol):
     # Testing purposes
     parent_class = WebSocketServerProtocol
 
+    # Defer helpers
+    def deferToThread(self, func, *args, **kwargs):
+        d = deferToThread(func, *args, **kwargs)
+        self._callbacks.append(d)
+
+        def f(result):
+            if d in self._callbacks:
+                self._callbacks.remove(d)
+            return result
+        d.addBoth(f)
```

Comment:
It gets called whether its a failure passed in, or result, and since I return it, its passed regardless to the next in the chain.